src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -434,12 +437,17 @@ private void CheckCaptured(Symbol variable, ParameterSymbol rangeVariableUnderly
         /// <param name="variable"></param>
         private void NoteCaptured(Symbol variable)
         {
-            if (variable.Kind != SymbolKind.RangeVariable || this.regionPlace == PreciseAbstractFlowPass<LocalState>.RegionPlace.Inside)
+            if (variable.Kind != SymbolKind.RangeVariable || this.regionPlace == RegionPlace.Inside)
             {
                 _capturedVariables.Add(variable);
             }
+
+            (IsInside ? _capturedInside : _capturedOutside).Add(variable);
```

Comment:
Should this be moved into the above if statement?

Reply:
I mean, shouldn't it be conditioned on it not being a RangeVariable?

Reply:
Also, looking at this code, it would appear that `_capturedVariables` already serves the same purpose as `_capturedInside`. If that is correct, we should only have to add support for `_capturedOutside`.

Reply:
`_capturesVariables` contains both inside *and* outside captures